docs/deployment-ux.md

```diff
@@ -0,0 +1,410 @@
+```
+
+This will mark the revision as active, and start launching tasks to meet
+the deployment configuration's constraints. In the case of a Daemon
+environment, the desired target is to have one copy of the task for
+every instance in the target Instance Group. For the initial deployment,
+MinHealthyPercent is ignored, since we're starting from a situation that
+already breaches MinHealthyPercent
+
+``` {.shell}
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS          DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Deploying       5         0         0            0        
+```
+
+As the deployment progresses:
+
+``` {.shell}
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS          DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Deploying       5         0         0            0        
+```
+
+``` {.shell}
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS          DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Deploying       5         3         3            3         
+```
+
+Failure:
+
+``` {.shell}
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS          DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Stuck           5         4         4            3 
+```
+
+Success:
+
+``` {.shell}
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS          DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Active          5         5         5            5         
+```
+
+Updating an environment
+-----------------------
+
+Typically, you'd update an environment in order to:
+
+-   update the task definition that should be running
+-   change the set of instances on which to run tasks
+-   change the deployment configuration (i.e. to change health
+    thresholds) (TODO: Maybe require new env for this?)
+
+### Updating the task definition
+
+If you don't specify all attributes, you must specify the revision you
+wish to base your changes on. This prevents potential conflicts if
+multiple callers attempt to update an environment at the same time.
+
+``` {.shell}
+$ aws ecs update-environment --environment SomeEnvironment/Prod --revision 1 --task-definition "log-daemon:2"
+{ "EnvironmentRevision": 1 }
+```
+
+At this point, nothing has actually happened to your cluster. You can
+see this by showing your environment's status:
+
+``` {.shell}
+$ aws ecs describe-environment-status --environment-name SomeEnvironment/Prod --table
+REVISION                  STATUS     DESIRED   CURRENT   UP-TO-DATE   AVAILABLE
+SomeEnvironment/Prod:1    Active     5         5         5            5        
```

Comment:
In addition to this view of progress, we may wish to show a clearer breakdown of the progress of the entire environment to its goal state. For example:

```
$ aws ecs describe-environments
ENVIRONMENT           STATUS                  PROGRESS
SomeEnvironment/Gamma Active                  5/5 tasks
SomeEnvironment/Prod  Deploying revision 2    1/5 tasks
```